### DISCUSSION TITLE
Idea about KeyCloak/SocialLogin etc. Waht are your thoughts?

### DISCUSSION THREAD

**Discussion by AndreSteenbergen** [AvaloniaUI/Avalonia#12919] [General]
At this moment I am thinking abount embedding a local WebAPI in the Avalonia App. Then create a requestURL with redirect to point to my self hosted API and start the system url using:
```
ProcessStartInfo pinfo = new ProcessStartInfo(requrl) {UseShellExecute = true };
            Process.Start(pinfo);
```
The browser will start and call the internal Avalonia App hosted Web API telling about the token to retrieve an Access Token. What are your thoughts? Is there a nuget package already doing stuff like this?

The only issue I have is the process running the authentication part will show my http response and it will stay there. Is there a way to automatically come back and make the avaloina app to be the topmost app again?

The pro about just calling the url and doing the system wide url handling is the user is probably already logged in to google/facebook/whatever. So the login procedure could be quite nice, only part I don't like is the open active tab in the browser.

Is it possible to auto close the tab and return to the app? What would your idea be about that?

**Comment by AndreSteenbergen** [comment]
The reqUrl will be the openid connect/ google login page/ whatever with a redirect to localhost:PORT

I know about `IdentityModel.OidcClient` That is a great module to get the requestURL. I meant the part after that. The IBrowser part and return the access tokens.

Is there a way to regsiter an own scheme like myapp://oauthredirect/TOKEN

**Reply by AndreSteenbergen** [reply]
So basically this part:

![image](https://github.com/AvaloniaUI/Avalonia/assets/7676790/75030c9d-b4dd-49b2-9572-014dff4f432b)

**Reply by maxkatz6** [reply]
> Is there a way to regsiter an own scheme like myapp://oauthredirect/TOKEN

Yes, but it's platform-specific.
https://www.xamarinhelp.com/uri-scheme/
https://stackoverflow.com/questions/19458803/xamarin-mac-url-scheme

But if you have client and server running on the same machine, it should be easier (at least on desktop) to use named pipes - https://learn.microsoft.com/en-us/dotnet/standard/io/how-to-use-named-pipes-for-network-interprocess-communication

Or web sockets from client to web server. Or http streaming - https://dev.to/iamabdul/websockets-and-http-streaming-in-net-5k2.
